ddtrace/appsec/iast/taint_sinks/_base.py

```diff
@@ -74,45 +84,163 @@ def report(cls, evidence_value="", sources=None):
                 return None
 
             frame_info = get_info_frame(CWD)
-            if frame_info:
-                file_name, line_number = frame_info
-
-                # Remove CWD prefix
-                if file_name.startswith(CWD):
-                    file_name = os.path.relpath(file_name, start=CWD)
-
-                if isinstance(evidence_value, (str, bytes, bytearray)):
-                    evidence = Evidence(value=evidence_value)
-                elif isinstance(evidence_value, (set, list)):
-                    evidence = Evidence(valueParts=evidence_value)
-                else:
-                    log.debug("Unexpected evidence_value type: %s", type(evidence_value))
-                    evidence = ""
-
-                if cls.is_not_reported(file_name, line_number):
-
-                    _set_metric_iast_executed_sink(cls.vulnerability_type)
-
-                    report = _context.get_item(IAST.CONTEXT_KEY, span=span)
-                    if report:
-                        report.vulnerabilities.add(
-                            Vulnerability(
-                                type=cls.vulnerability_type,
-                                evidence=evidence,
-                                location=Location(path=file_name, line=line_number, spanId=span.span_id),
-                            )
-                        )
+            if not frame_info:
+                return None
+
+            file_name, line_number = frame_info
+
+            # Remove CWD prefix
+            if file_name.startswith(CWD):
+                file_name = os.path.relpath(file_name, start=CWD)
+
+            if _is_evidence_value_parts(evidence_value):
+                evidence = Evidence(valueParts=evidence_value)
+            elif isinstance(evidence_value, (str, bytes, bytearray)):
+                evidence = Evidence(value=evidence_value)
+            else:
+                log.debug("Unexpected evidence_value type: %s", type(evidence_value))
+                evidence = ""
+
+            if not cls.is_not_reported(file_name, line_number):
+                # not not reported = reported
```

Comment:
what's this comment?

Reply:
Double negations are sometimes hard to understand in code so this really means that the if means "if reported".